### ISSUE TITLE
SonarCloud checks are flaky

### ISSUE THREAD

**Issue by hariso** [ConduitIO/conduit-kafka-connect-wrapper#106]
### Bug description

The SonarCloud checks are flaky. From time to time, they fail with the error below. The mentioned projectKey and organization properties are set (in the Maven POM) and the SONAR_TOKEN env. variable is set too. Re-running the failed SonarCloud job always helped.

```
Error:  Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar (default-cli) on project conduit-kafka-connect-wrapper: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator -> [Help 1]
Error: 
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error: 
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```

### Steps to reproduce

1. Open a PR
2. Wait for SonarCloud check

**Expected**
The SonarCloud check is always triggered

**Actual**
The SonarCloud check sometimes doesn't get triggered

### Version

none

**Comment by neovintage** [comment]
Conduit is going through some work on stability. That will change the interfaces. We'll want to fix this before the 0.3 milestone gets merged. Otherwise, the connector will break.

**Comment by neovintage** [comment]
Conduit is going through some work on stability. That will change the interfaces. We'll want to fix this before the 0.3 milestone gets merged. Otherwise, the connector will break.

**Comment by hariso** [comment]
I _think_ I resolved this (a few dependency upgrades piled up and I'm about to start working on the OpenCDC changes).

The GH actions analysing the code kept complaining about the Sonar configuration, even though it's definitely there. It felt like it simply wasn't recognized (even though it was done as in instructions: SONAR_TOKEN in the environment variables, project key, org ID, Sonar host in pom.xml or through command line).

It looks like we have the same issue as here: https://community.sonarsource.com/t/doc-update-sonarscanner-for-maven-does-not-automatically-picks-sonar-token-from-the-environment/63599

I moved all the parameters into the command line, and it looks like it's (more) successful.

**Comment by hariso** [comment]
Fixed as described above, through #109 and #117.

**Comment by hariso** [comment]
After some time, we got a few more dependabot PRs, and the tests started failing again.

The debug logs for the actions were clear: no secret set.

However, in a few of my previous PRs, everything was well. I re-ran the check on one of my open PRs, and it was OK.

After poking around, I pushed a dummy commit (added a comment) to a dependabot PR and it passed. Then I found what I was missing: secrets need to be set for dependabot separately ([link](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-encrypted-secrets-for-dependabot)). I had the secret set for Actions, and whenever I was the actor in a GitHub action it worked. Whenever dependabot only was involved, it failed.
